scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/Meta.scala

```diff
@@ -0,0 +1,82 @@
+package scala.meta
+package internal
+package semantic
+
+import java.nio.charset.Charset
+import org.scalameta.data._
+import scala.{Seq => _}
+import scala.collection.immutable.Seq
+import scala.meta.io._
+import scala.meta.internal.semantic.{schema => s}
+import scala.meta.{semantic => m}
+import scala.meta.{Dialect => mDialect}
+import scala.meta.inputs.{Input => mInput, Position => mPosition, Point => mPoint}
+
+package object meta {
+  implicit class XtensionDatabaseSchema(mdatabase: m.Database) {
+    def toSchema(sourcepath: Sourcepath): s.Database = {
+      val sentries = mdatabase.entries.map {
+        case (minput, m.Attributes(mdialect, mnames, mmessages, mdenots, msugars)) =>
+          object mDialect {
+            def unapply(mdialect: mDialect): Option[String] = {
+              val isStandard = scala.meta.Dialect.standards.exists(_._2 == mdialect)
```

Comment:
I'm not sure I like this `mdialect` vs. `mDialect` trick, took me a while to spot.